tensorflow/contrib/android/java/org/tensorflow/contrib/android/TensorFlowInferenceInterface.java

```diff
@@ -50,6 +56,117 @@
   private static final String TAG = "TensorFlowInferenceInterface";
   private static final String ASSET_FILE_PREFIX = "file:///android_asset/";
 
+  /**
+   * Copies resource to a cache file.
+   * @param  context context The context that contains the files
+   * @param  String filename  Filename--take note that it checks for .pb in asset case
+   * @param  boolean res Is it from res/raw
+   *
+   * @return File reference to cache file.
+   */
+  private File createCacheFile(Context context, String filename, boolean res)
+      throws IOException {
+    InputStream inputStream = null;
+    if (res) {
+      int resourceId = context.getResources().getIdentifier(filename, "raw", context.getPackageName());
+      if (resourceId == 0) {
+        return null;
+      }
+      inputStream = context.getResources().openRawResource(resourceId);
+    } else {
+      if (!filename.endsWith(".pb")) {
+        filename += ".pb";
```

Comment:
@andrewharp  For instance, in your demo since you use assets, you have the filename with the extension, but I use raw and it does not use the extension.  I am simply checking if they have it before I try to pull the file from assets.  What assumption are you referring to?

Reply:
Is it a requirement to not use an extension on files in raw?

The assumption this seems to be making is that any model loaded from assets will have a .pb extension, even if the calling site doesn't request it. For example, if I name my graph "model.bin_proto" for whatever reason it will now try to load "model.bin_proto.pb". I don't think we want to force this.

Reply:
Yes. Res/raw requires no extension while assets requires one. I am only making this worst case assumption: I passed in a model name without extension and now, if I cannot find it in raw, I want to ensure if it is in assets, I have added its extension to the filename passed in before I try to find it there.  Your demo code for instance:
private static final String MODEL_FILE = "file:///android_asset/rounded_graph.pb";

Reply:
Just to be clear, res/raw does not require an extension, or requires that the file has no extension? That is, is it possible to put a file with an extension in there?

Regarding loading from assets, it seems the caller should have the freedom to make the extension whatever they want and not be forced to use ".pb". Just because the examples use models ending with pb does not mean we should assume this is always the case.

Reply:
Res/raw file are accessible by resource id or filename without the extension.  I did not consider that files outside of .pb would be involved.  Protobufs used by tensorflow are usually binary and hold that extension.  I would have to do more research to find out if there is a use case for that; I tend to agree it is a bit brittle and can move that check outside of the interface.

Reply:
Yes, please remove the forced file extension and address the other comments and I think this is good to go.

Reply:
@andrewharp I am still testing it, but should be done shortly.  The changes to aapt make the res/raw access even more complicated, but there is a workaround.